### ISSUE TITLE
Many questions about BigNum

### ISSUE THREAD

**Issue by avently** [korlibs/kbignum#2]
Hello. First of all, thank you for creating multiplatform libs for Kotlin! I spent hours reading code on Github and documentation. It's useful for entire community.
Let's talk about this library.
- What version is the latest? With 0.46.0 I'm unable to use the lib because IDEA tells me about pre-release compiler version. With 0.9.0 it works fine.
- Do you want this lib to be public and used by other people? Because looks like you are not really want to make it as fast as possible.
A little comparsion between BigDecimal and BigNum shows it:

BigDecimal test 1: ended after 3.947937 ms
BigNum test 1: ended after 61.848497 ms

0.00
0.01
0.02
0.02
0.03
0.03
0.04
0.05
0.05
0.06
BigDecimal test 2: ended after 0.75538 ms

0
0
0
0
0
0
0
0
0
0
BigNum test 2: ended after 2.097193 ms

Code for tests: 
```
val bigDecimals = arrayListOf<BigDecimal>()
    val bigNums = arrayListOf<BigNum>()

    timeCounter(true, "BigDecimal test 1") {
        repeat(10) {
            bigDecimals.add(BigDecimal(it.toString()))
        }
    }

    timeCounter(true, "BigNum test 1") {
        repeat(10) {
            bigNums.add(BigNum(it.toString()))
        }
    }

    val bigDecimal = BigDecimal("173.0")
    val bigNum = BigNum("173.0")

    timeCounter(true, "BigDecimal test 2") {
        repeat(10) {
            println(bigDecimals[it].divide(bigDecimal, 2, RoundingMode.CEILING))
        }
    }

    timeCounter(true, "BigNum test 2") {
        repeat(10) {
            println(bigNums[it].div(bigNum, 2))
        }
    }
```

So seems like perfomance could be better. Also there is no working divide method and no any RoundingMode option. 
Please, tell me you plans for this lib because I really hope to find BigDecimal alternative for multiplatform configuration. I'm not sure that my knowledge in Math can help you...

**Comment by soywiz** [comment]
I have just published `0.10.0` here: https://bintray.com/soywiz/soywiz/kbignum

The implementation is slow. I have implemented it straightforward in a way that is simple. Optimizations should come later. But this project is not my main priority.

Also I wanted to use native big number libraries (BigInteger from JVM or bignum from JS when available) and that's not done yet.

If you can find a public-domain library or a MIT library, that would work too, I can port it and give the proper credit, if not, I have to figure out things myself to be able to provide a MIT license at least.
https://korlibs.soywiz.com/ktcc/ <-- this might help too to convert C libraries

**Comment by avently** [comment]
@soywiz another library I found that looks interesting for kotlin world is https://github.com/fs523577192/big_number

It's just a port of java implementation to kotlin common code. Didn't test it so maybe it's not working like it should.
But the idea is not so bad. Common BigInteger/BigDecimal everywhere with help of java developers is really useful thing.
